include/checker/zx/ZXChecker.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "Configuration.hpp"
+#include "Definitions.hpp"
+#include "EquivalenceCriterion.hpp"
+#include "QuantumComputation.hpp"
+#include "ZXDiagram.hpp"
+#include "checker/EquivalenceChecker.hpp"
+#include "nlohmann/json.hpp"
+
+namespace ec {
+    class ZXEquivalenceChecker: public EquivalenceChecker {
+    public:
+        ZXEquivalenceChecker(const qc::QuantumComputation& qc1, const qc::QuantumComputation& qc2, Configuration configuration) noexcept;
+
+        EquivalenceCriterion run() override;
+
+        void json(nlohmann::json& j) const noexcept override {
+            EquivalenceChecker::json(j);
+            j["checker"] = "zx";
+        }
+
+    private:
+        zx::ZXDiagram miter;
+        zx::fp        tolerance = zx::TOLERANCE;
```

Comment:
Is it purposely not intended to be able to change the tolerance for the ZX Package?
(As for the DD Package, we do allow to change that as part of the `numerical_tolerance` setting)

Reply:
It's totally fine with me to fix this tolerance for now. However, I would argue that there should not be two different values here and in the default value of the constructor.

Reply:
Also I just realized that `traceThreshold` of `0.0` means no distinct phases are considered equal no matter how close. I guess that could be the intention.